### ISSUE TITLE
Go2 Pathing Updates to exclude Popular Rooms

### ISSUE THREAD

**Issue by intangibles** [rpherbig/dr-scripts#1796]
Hey All,

I'm going to start updating ;go2 rooms to exclude popular rooms from go2 travel paths. 

I'll make the updates by changing the timeto on these rooms to something like 10 seconds.  It won't affect go2 pathing to these rooms, but it will make you detour.  For example I can update the timeto on the room outside the empath guild to 10 seconds, which will cause a 2-3 room detour and add possibly .6 seconds to the total travel time.  The detour will essentially be negligible and the bright side is people won't have to stare at those people running back and forth.

I won't update any rooms that will have a significant pathing affect or if there is clearly only one path through.

We can take this in other directions too, for example, we could update all timetos to crossing to force people to take the bramble path north of crossing when they are bypassing crossing.

For our purposes here, if people can post any popular rooms they would like to see bypassed please post them.  If you know any shortcuts or pathings that may be beneficial, lets post those too and we can discuss.

Thanks!

**Comment by rpherbig** [comment]
By 'popular rooms' do you mean populated rooms? Is this to try and be a bit more unobtrusive?

**Comment by intangibles** [comment]
@rpherbig 

Exactly.  There was complaining on discord earlier about "lich people scripting in Crossing" so I figured this could be some low hanging fruit.  I'm actually not sure if there are any super populated rooms that people run through currently, but might as well take a look.

**Comment by rpherbig** [comment]
If you're interested in investigating this further, cool. I think some people on Discord will complain no matter what we do. You might see if they have any specific complaints.

**Comment by jandersson** [comment]
Go2 isn't handled in this repository, nor is the mapdb. I disagree that you should modify it to please a handful of users.

**Comment by jandersson** [comment]
I also want to implore you not to change the data to something that is not truthful. You may have good intentions, but it will affect things you have not considered.

**Comment by rpherbig** [comment]
To clarify my position, I'm all for understanding what people are complaining about. I'm in agreement with @jandersson that this particular approach would have unanticipated side-effects; I think we should wait to determine a solution until we understand the problem better.

**Comment by intangibles** [comment]
@jandersson 
I think you are completely misinterpreting what I said. What I suggested is common place in EVERY single node, gathering spot, safe-rooom, and populated area in Gemstone.  It's just common courtesy not to walk through someone's room 100 times if you can avoid it.  Even if it improved one person's gameplay, that should be enough.

There is absolutely no modification of any data to something untruthful. I would appreciate if you could take a step back and understand what is going on before lecturing.

Let me expand.  TimeTo is an arbitrary number that we enter as players, or the script enters, to determine how long it takes to enter and leave a room. When go2 determines pathing it adds up all the values of timeto to determine the correct path to take.  It is literally impossible for this data to be untruthful because its an arbitrary value place on the passage of time through a room in order to determine the fastest route.  The default is currently 0.2 but it could be 10 and we would end up at the same result.

Changing a common area timeto from 0.2 to 5 only affects the pathing time THROUGH that room. If for some reason no alternate route is available or the time differential is greater then 5, it will pass through that room. It does not affect going TO the room at all.

And you are right, this has nothing to do with the repository. I don't need anyone's input to do it. Hell, I could have simply made the change and I doubt anyone would have noticed not passing by the empath hall. Hell, for the longest time go2 wasn't passing through the bridge and no one seemed to notice when I fixed it.  This is literally the exact same situation.

Even if this is not repository related, the majority of lich players are using it so this is a good place to discuss.

**Comment by rcuhljr** [comment]
``` I doubt anyone would have noticed not passing by the empath hall.```

I guess my question is, why? The front door of the empath room isn't a real gathering spot, and lich travels basically the same paths genie and everyone else uses. I just sat there and recorded everyone who came by during the course of a tv show and lich users were the minority. (less than 20%) And not once did anyone stop or speak or even train in the room.

All we'd do is draw even more attention to lich by driving traffic through odd rooms that aren't normally traveled.  

Do we have an example of an area where this is really a problem? The only thing we've heard tangential to this is people who purposefully choose training rooms in crowded areas with travel prone setups, but this change wouldn't fix that. 

It seems like there's better places to focus energy currently then on map edits to address ill defined problem.

**Comment by intangibles** [comment]
@rcuhljr 

You can't necessarily measure it like that. Sometimes outside the empath hall is a gathering place.  If it is, we should limit the amount of traffic running by.  Same with Pfanstons grove as an example.   Even if its a 20% reduction, why not? The amount of effort is extremely tiny. I actually used to reroute around that area with my genie scripts.

Anyways, the purpose of this initial post was to determine if there were any areas that may need this. If there are none, great.

**Comment by jandersson** [comment]
> I would appreciate if you could take a step back and understand what is going on before lecturing.

I don't think I've lectured you in my reply - I understand what you want to do, I am disagreeing with it. Timeto isn't arbitrary, its an estimated cost of the movement in seconds. When you change this to be an incorrect value, the shortest path algorithm is no longer giving you a shortest path.  Feel free to change it, but in my opinion it goes against the spirit of what that data is supposed to represent.

**Comment by intangibles** [comment]
I'm going to close this issue.

I've been keeping an eye on this since I made the issue.   This is largely moot.  There are no popular rooms that go2 paths over that would need this type of treatment. (I'm not even sure there is any gathering rooms on any paths for that matter.)

Side-note, I did notice an attunement room in Lang base-towns was perceiving in the bin room. This would be an example of a room we would want to phase out.  (People kept trying to heal me and i would immediately leave.)
